projects/client-certificate-authentication.md

```diff
@@ -0,0 +1,112 @@
+# Use Client Certificate Authentication
+
+Here are the steps to perform user authentication using X509 client certificate authentication on an Android emulator:
+
+## Step 1: Import the self-signed CA file to the keychain of your local machine
+
+* Double click on the `certificate.crt` file that is located in the `./projects/certs/ca` directory to import it into the keychain. Make sure the CA certificate is trusted.
+
+## Step 2: Create user certificates and add it to the trust keystore
+
+* Run the following commands to create the user certificate:
+
+  ```
+  cd ./projects/certs
+
+  # Generate the client key & cert. You will be prompted for some questions about the certificate. Make sure the answer to `What is your first and last name` is the user's email address.
+  keytool -genkey -alias client-alias -keyalg RSA -keypass changeit  -storepass changeit -keystore client_keystore.jks
+  ```
+
+* Export the user's certificate and add it to the trust store.
+
+  ```
+  # Export the user cert
+  keytool -export -alias client-alias -storepass changeit -file client.cer -keystore client_keystore.jks
+
+  # Import the user cert into the trust.keystore file. You should change the alias value to the user's name
+  cd ../keycloak
+  keytool -import -v -trustcacerts -alias user1 -file ../certs/client.cer -keystore trust.keystore -keypass Password1 -storepass Password1
+  ```
+
+* Distribute the client certificate
+
+  To make it easier to distribute the client cert, we convert it to PKCS12 format first:
+
+  ```
+  cd ./projects/certs
+  keytool -importkeystore -srckeystore client_keystore.jks -destkeystore client_keystore.p12 -deststoretype PKCS12
```

Comment:
Maybe add a reminder to what the storepass password is here (`changeit`) and it might cause confusion with the `Password1` given in the last command above.